user-operator/src/main/java/io/strimzi/operator/user/operator/SimpleAclOperator.java

```diff
@@ -190,84 +200,72 @@ public SimpleAclOperator(Vertx vertx, kafka.security.auth.SimpleAclAuthorizer au
      */
     public Set<SimpleAclRule> getAcls(String username)   {
         log.debug("Searching for ACL rules of user {}", username);
-        Set<SimpleAclRule> result = new HashSet<SimpleAclRule>();
+        Set<SimpleAclRule> result = new HashSet<>();
         KafkaPrincipal principal = new KafkaPrincipal("User", username);
 
-        scala.collection.immutable.Map<Resource, scala.collection.immutable.Set<Acl>> rules;
+        AclBindingFilter aclBindingFilter = new AclBindingFilter(ResourcePatternFilter.ANY,
+            new AccessControlEntryFilter(principal.toString(), null, AclOperation.ANY, AclPermissionType.ANY));
 
+        Collection<AclBinding> aclBindings = null;
         try {
-            rules = authorizer.getAcls(principal);
-        } catch (Exception e)   {
-            log.error("Failed to get existing Acls rules for user {}", username, e);
-            throw e;
+            aclBindings = adminClient.describeAcls(aclBindingFilter).values().get();
+        } catch (InterruptedException | ExecutionException e) {
+            // Admin Client API needs authorizer enabled on the Kafka brokers
+            if (e.getCause() instanceof SecurityDisabledException) {
+                throw new InvalidResourceException("Authorization needs to be enabled in the Kafka custom resource", e.getCause());
+            } else if (e.getCause() instanceof UnknownServerException) {
```

Comment:
Do we need to check the message here to make sure this does not hide some other exception?

Reply:
Taking a look at Kafka codebase it could make sense because there are other scenarios raising this kind of exception.